templates/optionalFeatures/test-cafe/testcafeSetup.js

```diff
@@ -0,0 +1,90 @@
+const inquirer = require("inquirer");
+const testcafeConfig = require("./testcaferc.json");
+
+const { readdirSync } = require("fs");
+
+const exec = require("child_process").exec;
+
+const getDirectories = (source) =>
+  readdirSync(source, { withFileTypes: true })
+    .filter((dirent) => dirent.isDirectory())
+    .map((dirent) => dirent.name);
+
+const APP_TYPE = getDirectories("./apps");
+var fs = require("fs");
+
+const addTestcafeSetup = (envVars) => {
+  const dir = `./apps/web/${envVars.APP_NAME}/e2e-${envVars.APP_NAME}/tests`;
+  if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir, {
+      recursive: true,
+    });
+    fs.mkdirSync(`${dir}/models/stepDefs`, { recursive: true });
```

Comment:
Keeps a basic directory template in optional features and copy in required destination. No need to create folders here.